### PR TITLE
[PWA-1709] Consolidate and use WishlistButton

### PR DESCRIPTION
**Pull request by tjwiebell**
## Description

As a user I want a consistent UX with regards to wishlists.

As a developer I want a consistent component / coding experience.

Preconditions

Wishlist is enabled in Magento Admin Panel > Stores > Settings > Configuration > Customers > Wish List
Multi-wishlist is not enabled in Stores > Settings > Configuration > Customers > Wish List > Enable Multiple Wish Lists (EE) OR Maximum number of wishlists is set to one (1)
Acceptance Criteria

Login toasts are rendered if user clicks to add to wishlist but is not signed in
If EE only one wishlist is allowed, no dialog (to choose a wishlist) should appear
The "general" Wishlist Button component takes props such as localized "buttonText" string for customization
The "general" Wishlist Button component takes props for beforeAdd and afterAdd callbacks

The goal of this story is to replace any other UX/logic with the existing WishlistButton component, with some changes such as:

- Rendering a login toast if user clicks to add but is not signed in.
- Modify EE functionality to act as CE if only 1 wishlist is allowed (no dialog popup)
- Take props such as "buttonText" string
- Take props for beforeAdd and afterAdd callbacks

## Related Issue

* [[PWA-1709](https://jira.corp.magento.com/browse/PWA-1709)] Consolidate and use WishlistButton

#### Test scenario(s) for direct fix/feature
1. Navigate to a category page
2. Verify that the wishlist button is present on each product tile
3. Verify that when the user is not signed in, clicking the wishlist button pops a "you need to login first" toast
4. Verify that once the user is signed in, clicking the wishlist button adds the product to the single wishlist. No dialog to choose which wishlist appears (based on Preconditions).
5. Navigate to a product page
6. Verify that the wishlist button is always present (signed in and not)
7. Verify that when the user is not signed in, clicking the wishlist button pops a "you need to login first" toast
8. Verify that once the user is signed in, clicking the wishlist button adds the product to the single wishlist. No dialog to choose which wishlist appears ((based on Preconditions).
9. Add an item to Cart and navigate to /cart page
10. Verify that when the user is not signed in, clicking the wishlist button pops a "you need to login first" toast
11. Verify that once the user is signed in, clicking the wishlist button adds the product to the single wishlist. No dialog to choose which wishlist appears (based on Preconditions).

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by supernova-at — APPROVED** [review]
A couple very minor things that you may want to update, but overall 👍 .

Verification steps passed ✅

File: packages/peregrine/lib/talons/LegacyMiniCart/useKebab.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const useKebab = () => {
         }
     }, []);
 
-    useEventListener(globalThis.document, 'mousedown', handleOutsideKebabClick);
+    useEventListener(globalThis.document, 'mouseup', handleOutsideKebabClick);
```
[the remark this answers is not in the document]
Comment (tjwiebell) [reply packages/peregrine/lib/talons/LegacyMiniCart/useKebab.js:20]: It was preventing the `addToListButton.onClick` event from firing since it would unmount on mousedown.

File: packages/peregrine/lib/talons/Wishlist/AddToListButton/helpers/useSingleWishlist.js [hunk 2]
```diff
@@ -125,6 +135,7 @@ export const useSingleWishlist = props => {
 
     return {
         buttonProps,
+        buttonText: props.buttonText && props.buttonText(isSelected),
```
[the remark this answers is not in the document]
Comment (tjwiebell) [reply packages/peregrine/lib/talons/Wishlist/AddToListButton/helpers/useSingleWishlist.js:138]: This was my attempt at a render prop, but this change might make sense. Right now the heart will immediately fill in, but the text won't change until the mutation completes. UI might as well be optimistic, so I think I'll make this change.

File: packages/venia-ui/lib/components/CartPage/ProductListing/product.css [hunk 3]
```diff
@@ -107,6 +107,20 @@
...
+    column-gap: 1.125rem;
+    display: inline-flex;
+    font-size: 0.875rem;
+    padding: 0.875rem 0.625rem;
+    width: 100%;
+}
+
+.addToListButton_selected {
+    display: none;
+}
```
[the remark this answers is not in the document]
Comment (tjwiebell) [reply packages/venia-ui/lib/components/CartPage/ProductListing/product.css:122]: The "Save for later" option won't show up in the kebab menu if already in favorites. UX didn't like the idea of a selected option in the kebab you couldn't interact with.

File: packages/venia-ui/lib/components/CartPage/ProductListing/product.css [hunk 4]
```diff
@@ -107,6 +107,20 @@
     display: none;
 }
 
+.addToListButton {
+    --stroke: rgb(var(--venia-global-color-teal));
+    align-content: center;
+    column-gap: 1.125rem;
+    display: inline-flex;
+    font-size: 0.875rem;
```
[the remark this answers is not in the document]
Comment (tjwiebell) [reply packages/venia-ui/lib/components/CartPage/ProductListing/product.css]: Dropped some tokens into this file where possible in 68756d0.

**Review by supernova-at — APPROVED** [review]
